Developer can't clone + run make [JIRA: RIAK-1653]

Commands run:

```
git clone -o basho git@github.com:basho/webmachine.git
cd webmachine
source /home/erlang/installs/OTP_R16B02_basho7/activate
make
```

Output:

```
(OTP_R16B02_basho7)lbakken@brahms ~/Projects/basho/webmachine (develop=)
$ make
touch: cannot touch ‘./.rebar/DEV_MODE’: No such file or directory
Makefile:42: recipe for target 'DEV_MODE' failed
make: *** [DEV_MODE] Error 1
```

Expected:

Successful compilation.

Workaround: create `.rebar` directory first.

```
mkdir .rebar
```

--- a/Makefile
+++ b/Makefile
@@ -20,7 +20,7 @@ clean:
 # nuke deps first to avoid wasting time having rebar recurse into deps
 # for clean
 distclean:
-	@rm -rf deps ./rebar/DEV_MODE
+	@rm -rf ./deps ./.rebar
 	@(./rebar clean)
 
 edoc:
@@ -39,4 +39,5 @@ travisupload:
 	travis-artifacts upload --path ${ARTIFACTSFILE}
 
 DEV_MODE:
+	@[ -d ./.rebar ] || mkdir ./.rebar
 	@touch ./.rebar/DEV_MODE

